Remove String::stripWhiteSpace()

annevk opened the pull request:
#### 4b6eaec45e978ed59f9161c88f8e978ad7fba925
<pre>
Remove String::stripWhiteSpace()
<a href="https://bugs.webkit.org/show_bug.cgi?id=257191">https://bugs.webkit.org/show_bug.cgi?id=257191</a>
rdar://109703510

Reviewed by Darin Adler.

The name is too generic as there are many flavors of whitespace and it
used a deprecated whitespace variant to boot.

This changes deprecatedIsSpaceOrNewline, its inverse, and
isUnicodeWhitespace to use UChar. This is more compatible with the
various templates and White_Space does not go beyond U+3000 (and if it
ever did go beyond UChar we&apos;d have to change the callers somehow anyway
to account for that).

This change exports StringImpl::stripLeadingAndTrailingCharacters() so
it can be used by XPathGrammar.cpp.

This change updates XPathGrammar.y with some changes that were made to
XPathGrammar.cpp directly. Unfortunately I was not able to backport all
changes. <a href="https://bugs.webkit.org/show_bug.cgi?id=257189">https://bugs.webkit.org/show_bug.cgi?id=257189</a> tracks
completing that work.

* Source/WTF/wtf/text/StringImpl.cpp:
(WTF::StringImpl::stripWhiteSpace): Deleted.
* Source/WTF/wtf/text/StringImpl.h:
(WTF::deprecatedIsSpaceOrNewline):
(WTF::isUnicodeWhitespace):
(WTF::deprecatedIsNotSpaceOrNewline):

Make it more clear that isUnicodeWhitespace is not deprecated by
separating it out.

* Source/WTF/wtf/text/WTFString.cpp:
(WTF::String::stripWhiteSpace const): Deleted.
* Source/WTF/wtf/text/WTFString.h:
* Source/WebCore/Modules/applicationmanifest/ApplicationManifestParser.cpp:
(WebCore::ApplicationManifestParser::parseGenericString):
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::computeAccessibilityIsIgnored const):
(WebCore::AccessibilityNodeObject::textUnderElement const):

Use isAllSpecialCharacters instead as that&apos;s more efficient.

* Source/WebCore/css/parser/CSSParserImpl.cpp:
(WebCore::CSSParserImpl::consumeSupportsRule):
* Source/WebCore/dom/ImageOverlay.cpp:
(WebCore::ImageOverlay::updateSubtree):

Use StringViews for efficiency.

* Source/WebCore/dom/ScriptElement.cpp:
(WebCore::ScriptElement::determineScriptType const):
* Source/WebCore/editing/TextManipulationController.cpp:
(WebCore::areEqualIgnoringLeadingAndTrailingWhitespaces):

Inline this comparison for now as there&apos;s only one and use StringView
for efficiency.

* Source/WebCore/editing/TypingCommand.cpp:
(WebCore::TypingCommand::markMisspellingsAfterTyping):
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::mediaSessionTitle const):
* Source/WebCore/html/HTMLOptGroupElement.cpp:
(WebCore::HTMLOptGroupElement::groupLabelText const):
* Source/WebCore/html/ValidatedFormListedElement.cpp:
(WebCore::ValidatedFormListedElement::updateVisibleValidationMessage):
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::selectorsFromSource):
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.cpp:
(WebCore::checkMediaType):

Use StringView for efficiency and add a FIXME for additional possible
cleanup.

* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::truncatedStringForMenuItem):
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::isMediaDiskCacheDisabled):

Add a FIXME to indicate this could probably be improved.

* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp:
(WebCore::SourceBufferPrivateGStreamer::platformMaximumBufferSize const):
* Source/WebCore/platform/gtk/SelectionData.cpp:
(WebCore::SelectionData::setURIList):
* Source/WebCore/platform/ios/DragImageIOS.mm:
(WebCore::createDragImageForLink):
* Source/WebCore/platform/mac/PasteboardMac.mm:
(WebCore::writeURLForTypes):
(WebCore::Pasteboard::writeTrustworthyWebURLsPboardType):
* Source/WebCore/platform/mac/PasteboardWriter.mm:
(WebCore::createPasteboardWriter):

Add a FIXME to indicate this could probably be improved.

* Source/WebCore/platform/network/MIMEHeader.cpp:
(WebCore::retrieveKeyValuePairs):
(WebCore::MIMEHeader::parseHeader):
* Source/WebCore/platform/network/ParsedContentType.cpp:
(WebCore::ParsedContentType::setContentType):
* Source/WebCore/platform/network/curl/CookieUtil.cpp:
(WebCore::CookieUtil::parseCookieAttributes):
(WebCore::CookieUtil::parseCookieHeader):
* Source/WebCore/platform/network/curl/CurlMultipartHandle.cpp:
(WebCore::CurlMultipartHandle::extractBoundary):
* Source/WebCore/platform/network/curl/CurlRequest.cpp:
(WebCore::CurlRequest::didReceiveDebugInfo):
* Source/WebCore/platform/network/curl/ResourceResponseCurl.cpp:
(WebCore::ResourceResponse::appendHTTPHeaderField):
* Source/WebCore/platform/text/PlatformLocale.cpp:
(WebCore::Locale::convertFromLocalizedNumber):
* Source/WebCore/platform/win/ClipboardUtilitiesWin.cpp:
(WebCore::extractMarkupFromCFHTML):
(WebCore::fragmentFromCFHTML):
* Source/WebCore/rendering/RenderMenuList.cpp:
(RenderMenuList::setTextFromOption):
* Source/WebCore/svg/properties/SVGAnimationAdditiveValueFunctionImpl.cpp:
(WebCore::SVGAnimationColorFunction::calculateDistance const):
* Source/WebCore/svg/properties/SVGPropertyTraits.h:
(WebCore::SVGPropertyTraits&lt;Color&gt;::fromString):
(WebCore::SVGPropertyTraits&lt;Color&gt;::parse):
* Source/WebCore/xml/XPathGrammar.cpp:
* Source/WebCore/xml/XPathGrammar.y:
* Source/WebKit/UIProcess/API/glib/WebKitOptionMenuItemPrivate.h:
(_WebKitOptionMenuItem::_WebKitOptionMenuItem):
* Source/WebKit/UIProcess/gtk/WebPopupMenuProxyGtk.cpp:
(WebKit::WebPopupMenuProxyGtk::createPopupMenu):
* Source/WebKit/WebProcess/WebCoreSupport/WebContextMenuClient.cpp:
(WebKit::WebContextMenuClient::searchWithGoogle):
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebContextMenuClientMac.mm:
(WebKit::WebContextMenuClient::searchWithGoogle):
* Tools/TestWebKitAPI/Tests/WebCore/DisplayListRecorderTests.cpp:

Canonical link: <a href="https://commits.webkit.org/264468@main">https://commits.webkit.org/264468@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/99d836d98180d8fe9347a5a5ad731f9a56ee26ec

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7717 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7988 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8169 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9358 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7874 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/7723 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/9978 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7909 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10745 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7851 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/8967 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/7086 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9466 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6277 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/7026 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14692 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/6580 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/7433 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/7150 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10558 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/7300 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7640 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/6228 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/7872 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6968 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1803 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1841 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11179 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/8084 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7374 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1952 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/text/StringImpl.h:
```diff
@@ -629,11 +628,11 @@ int codePointCompare(const StringImpl*, const StringImpl*);
-bool deprecatedIsSpaceOrNewline(UChar32);
-bool isUnicodeWhitespace(UChar32);
+bool deprecatedIsSpaceOrNewline(UChar);
+bool isUnicodeWhitespace(UChar);
```
It is not so great that this is paragraphed in with the comment about why the other function is deprecated. A blank line would help.

I understand why this was changed to take a `UChar`, since it can be passed to functions that want a code unit character, but it is now dangerous if anyone calls it with a `UChar32`, because it will chop off the lower 16 bits and possibly give the wrong result.

An alternative would be a function with the words “code unit” in its name that is intended to pass to code unit predicates; the main function can be overloaded for different types for efficiency and correctness. Or we can just be careful to never call this with `UChar32` values.

annevk replied:
Being careful seems okay to me. I tried using the same kind of template we have for `isASCIIWhitespace`, but that does not appear to be compatible for some reason. But perhaps it is and I just need to update most call sites to use `deprecatedIsSpaceOrNewline<UChar>`? As I'm relatively new to this I still get compiler feedback confused at times.

On the other hand, if that is worth doing we should probably only do it for `isUnicodeWhitespace` as we don't want the deprecated one to end up in new code anyway.

darinadler commented on Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
```diff
@@ -580,7 +580,7 @@ bool AccessibilityNodeObject::computeAccessibilityIsIgnored() const
 
         // Whitespace only text elements should be ignored when they have no renderer.
-        String string = stringValue().stripWhiteSpace().simplifyWhiteSpace();
+        String string = stringValue().stripLeadingAndTrailingCharacters(deprecatedIsSpaceOrNewline).simplifyWhiteSpace();
```
Since this call site is just checking if all characters are “spaces”:

1. There is no need to call `simplifyWhiteSpace` since that never turns a non-empty string into an empty string.
2. There is no need to use the `String` function that allocates memory instead of the `StringView` function that is more efficient since it does not.
3. It could use a function that returns a `bool` instead of a `String`. I think we have such functions on `StringView`, named `containsOnly` perhaps.

annevk replied:
Nice! Done. Also filed https://bugs.webkit.org/show_bug.cgi?id=257251 as follow-up as `containsOnly` seems like a better name.

darinadler commented on Source/WebCore/dom/ImageOverlay.cpp:
```diff
@@ -346,7 +346,7 @@ static Elements updateSubtree(HTMLElement& element, const TextRecognitionResult&
 
                 for (size_t childIndex = 0; childIndex < childResults.size(); ++childIndex) {
-                    if (childResults[childIndex].text != childTextElements[childIndex]->textContent().stripWhiteSpace())
+                    if (childResults[childIndex].text != childTextElements[childIndex]->textContent().stripLeadingAndTrailingCharacters(deprecatedIsSpaceOrNewline))
```
Since the only use here of the result is to compare with another string this could use the `StringView` version for better efficiency.

darinadler commented on Source/WebCore/dom/ImageOverlay.cpp:
```diff
@@ -358,7 +358,7 @@ static Elements updateSubtree(HTMLElement& element, const TextRecognitionResult&
                         return false;
 
-                    if (textContentByLine[lineIndex++].stripWhiteSpace() != text.wholeText().stripWhiteSpace())
+                    if (textContentByLine[lineIndex++].stripLeadingAndTrailingCharacters(deprecatedIsSpaceOrNewline) != text.wholeText().stripLeadingAndTrailingCharacters(deprecatedIsSpaceOrNewline))
```
Since the only use here of the result is to compare with another string this could use the `StringView` version for better efficiency.

darinadler commented on Source/WebCore/editing/TextManipulationController.cpp:
```diff
@@ -289,7 +289,7 @@ static bool canPerformTextManipulationByReplacingEntireTextContent(const Element
 static bool areEqualIgnoringLeadingAndTrailingWhitespaces(const String& content, const String& originalContent)
 {
-    return content.stripWhiteSpace() == originalContent.stripWhiteSpace();
+    return content.stripLeadingAndTrailingCharacters(deprecatedIsSpaceOrNewline) == originalContent.stripLeadingAndTrailingCharacters(deprecatedIsSpaceOrNewline);
```
Since the only use here of the result is to compare with another string this could use the `StringView` version for better efficiency. The cleanest way to do that here is probably to change the types of the two function arguments.

darinadler commented on Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.cpp:
```diff
@@ -132,7 +132,7 @@ static inline bool checkMediaType(ContentSecurityPolicyMediaListDirective* direc
     if (!directive)
         return true;
-    if (typeAttribute.isEmpty() || typeAttribute.stripWhiteSpace() != type)
+    if (typeAttribute.isEmpty() || typeAttribute.stripLeadingAndTrailingCharacters(deprecatedIsSpaceOrNewline) != type)
```
Since the only use here of the result is to compare with another string this could use the `StringView` version for better efficiency. The cleanest way to do that here is probably to change the type of the `typeAttribute` function argument.

annevk replied:
I left a FIXME for changing the argument type as it's quite a deep call stack.

darinadler commented on Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
```diff
@@ -2786,7 +2786,7 @@ bool isMediaDiskCacheDisabled()
         auto s = String::fromLatin1(std::getenv("WPE_SHELL_DISABLE_MEDIA_DISK_CACHE"));
         if (!s.isEmpty()) {
-            String value = s.stripWhiteSpace().convertToLowercaseWithoutLocale();
+            String value = s.stripLeadingAndTrailingCharacters(deprecatedIsSpaceOrNewline).convertToLowercaseWithoutLocale();
```
This could be rewritten to use `StringView` and `equalLettersIgnoringASCIICase` instead of `convertToLowercaseWithoutLocale` to avoid memory allocation and use the simpler case-ignoring algorithm. There is no need to use `String` here at all for a C string that we are checking against ASCII constants. In fact I don’t know why WTF is even used instead of functions like `strcmp`. Very unclear why it’s even desirable to strip spaces from an environment variable value. But I have rarely helped with the gstreamer code and maybe there is something I am missing. Could just leave this as is.

annevk replied:
I left a FIXME.

darinadler commented on Source/WebCore/platform/mac/PasteboardWriter.mm:
```diff
@@ -81,7 +81,7 @@
 
         // WebURLsWithTitlesPboardType.
-        auto paths = adoptNS([[NSArray alloc] initWithObjects:@[ @[ cocoaURL.absoluteString ] ], @[ urlData->title.stripWhiteSpace() ], nil]);
+        auto paths = adoptNS([[NSArray alloc] initWithObjects:@[ @[ cocoaURL.absoluteString ] ], @[ urlData->title.stripLeadingAndTrailingCharacters(deprecatedIsSpaceOrNewline) ], nil]);
```
Could use `StringView` here to save an allocation in the rare case that we do strip characters. Would need to use thr explicit `StringView` function that creates an `NSString` since the convenient automatic conversion is built into the older `String` and not the newer `StringView`.

annevk replied:
I left a FIXME.

darinadler commented on Source/WebCore/xml/XPathGrammar.cpp:
```diff
@@ -1822,7 +1822,7 @@ YYSTYPE yylval;
         auto stringImpl = adoptRef((yyvsp[(3) - (4)].string));
         if (stringImpl)
-            stringImpl = stringImpl->stripWhiteSpace();
+            stringImpl = stringImpl->stripLeadingAndTrailingCharacters(deprecatedIsSpaceOrNewline);
```
Are we editing this generated file directly instead of regenerating it? I guess maybe someone made that error in the past and you are partially, but not entirely, fixing that here.

annevk replied:
That's correct. I linked a follow-up bug from the commit message.

darinadler commented on Source/WebCore/xml/XPathGrammar.y:
```diff
@@ -260,8 +260,10 @@ NodeTest:
+        auto stringImpl = adoptRef($3);
+        if (stringImpl)
+            stringImpl = stringImpl->stripLeadingAndTrailingCharacters(deprecatedIsSpaceOrNewline);
+        $$ = new WebCore::XPath::Step::NodeTest(WebCore::XPath::Step::NodeTest::ProcessingInstructionNodeTest, stringImpl.get());
```
Maybe we can move the stripping inside the `NodeTest` constructor.

annevk replied:
I wasn't really sure how to go about that. The only branching on `ProcessingInstructionNodeTest` happens in `nodeMatchesBasicTest()`.

darinadler reviewed (approved):
This is all good as is. I also mentioned a few places where we could use `StringView` and one place we can delete a call to `simolifyWhiteSpace`.

annevk committed:
Remove String::stripWhiteSpace()
https://bugs.webkit.org/show_bug.cgi?id=257191
rdar://109703510

Reviewed by Darin Adler.

The name is too generic as there are many flavors of whitespace and it
used a deprecated whitespace variant to boot.

This changes deprecatedIsSpaceOrNewline, its inverse, and
isUnicodeWhitespace to use UChar. This is more compatible with the
various templates and White_Space does not go beyond U+3000 (and if it
ever did go beyond UChar we'd have to change the callers somehow anyway
to account for that).

This change exports StringImpl::stripLeadingAndTrailingCharacters() so
it can be used by XPathGrammar.cpp.

This change updates XPathGrammar.y with some changes that were made to
XPathGrammar.cpp directly. Unfortunately I was not able to backport all
changes. https://bugs.webkit.org/show_bug.cgi?id=257189 tracks
completing that work.

* Source/WTF/wtf/text/StringImpl.cpp:
(WTF::StringImpl::stripWhiteSpace): Deleted.
* Source/WTF/wtf/text/StringImpl.h:
(WTF::deprecatedIsSpaceOrNewline):
(WTF::isUnicodeWhitespace):
(WTF::deprecatedIsNotSpaceOrNewline):

Make it more clear that isUnicodeWhitespace is not deprecated by
separating it out.

* Source/WTF/wtf/text/WTFString.cpp:
(WTF::String::stripWhiteSpace const): Deleted.
* Source/WTF/wtf/text/WTFString.h:
* Source/WebCore/Modules/applicationmanifest/ApplicationManifestParser.cpp:
(WebCore::ApplicationManifestParser::parseGenericString):
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::computeAccessibilityIsIgnored const):
(WebCore::AccessibilityNodeObject::textUnderElement const):

Use isAllSpecialCharacters instead as that's more efficient.

* Source/WebCore/css/parser/CSSParserImpl.cpp:
(WebCore::CSSParserImpl::consumeSupportsRule):
* Source/WebCore/dom/ImageOverlay.cpp:
(WebCore::ImageOverlay::updateSubtree):

Use StringViews for efficiency.

* Source/WebCore/dom/ScriptElement.cpp:
(WebCore::ScriptElement::determineScriptType const):
* Source/WebCore/editing/TextManipulationController.cpp:
(WebCore::areEqualIgnoringLeadingAndTrailingWhitespaces):

Inline this comparison for now as there's only one and use StringView
for efficiency.

* Source/WebCore/editing/TypingCommand.cpp:
(WebCore::TypingCommand::markMisspellingsAfterTyping):
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::mediaSessionTitle const):
* Source/WebCore/html/HTMLOptGroupElement.cpp:
(WebCore::HTMLOptGroupElement::groupLabelText const):
* Source/WebCore/html/ValidatedFormListedElement.cpp:
(WebCore::ValidatedFormListedElement::updateVisibleValidationMessage):
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::selectorsFromSource):
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.cpp:
(WebCore::checkMediaType):

Use StringView for efficiency and add a FIXME for additional possible
cleanup.

* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::truncatedStringForMenuItem):
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::isMediaDiskCacheDisabled):

Add a FIXME to indicate this could probably be improved.

* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp:
(WebCore::SourceBufferPrivateGStreamer::platformMaximumBufferSize const):
* Source/WebCore/platform/gtk/SelectionData.cpp:
(WebCore::SelectionData::setURIList):
* Source/WebCore/platform/ios/DragImageIOS.mm:
(WebCore::createDragImageForLink):
* Source/WebCore/platform/mac/PasteboardMac.mm:
(WebCore::writeURLForTypes):
(WebCore::Pasteboard::writeTrustworthyWebURLsPboardType):
* Source/WebCore/platform/mac/PasteboardWriter.mm:
(WebCore::createPasteboardWriter):

Add a FIXME to indicate this could probably be improved.

* Source/WebCore/platform/network/MIMEHeader.cpp:
(WebCore::retrieveKeyValuePairs):
(WebCore::MIMEHeader::parseHeader):
* Source/WebCore/platform/network/ParsedContentType.cpp:
(WebCore::ParsedContentType::setContentType):
* Source/WebCore/platform/network/curl/CookieUtil.cpp:
(WebCore::CookieUtil::parseCookieAttributes):
(WebCore::CookieUtil::parseCookieHeader):
* Source/WebCore/platform/network/curl/CurlMultipartHandle.cpp:
(WebCore::CurlMultipartHandle::extractBoundary):
* Source/WebCore/platform/network/curl/CurlRequest.cpp:
(WebCore::CurlRequest::didReceiveDebugInfo):
* Source/WebCore/platform/network/curl/ResourceResponseCurl.cpp:
(WebCore::ResourceResponse::appendHTTPHeaderField):
* Source/WebCore/platform/text/PlatformLocale.cpp:
(WebCore::Locale::convertFromLocalizedNumber):
* Source/WebCore/platform/win/ClipboardUtilitiesWin.cpp:
(WebCore::extractMarkupFromCFHTML):
(WebCore::fragmentFromCFHTML):
* Source/WebCore/rendering/RenderMenuList.cpp:
(RenderMenuList::setTextFromOption):
* Source/WebCore/svg/properties/SVGAnimationAdditiveValueFunctionImpl.cpp:
(WebCore::SVGAnimationColorFunction::calculateDistance const):
* Source/WebCore/svg/properties/SVGPropertyTraits.h:
(WebCore::SVGPropertyTraits<Color>::fromString):
(WebCore::SVGPropertyTraits<Color>::parse):
* Source/WebCore/xml/XPathGrammar.cpp:
* Source/WebCore/xml/XPathGrammar.y:
* Source/WebKit/UIProcess/API/glib/WebKitOptionMenuItemPrivate.h:
(_WebKitOptionMenuItem::_WebKitOptionMenuItem):
* Source/WebKit/UIProcess/gtk/WebPopupMenuProxyGtk.cpp:
(WebKit::WebPopupMenuProxyGtk::createPopupMenu):
* Source/WebKit/WebProcess/WebCoreSupport/WebContextMenuClient.cpp:
(WebKit::WebContextMenuClient::searchWithGoogle):
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebContextMenuClientMac.mm:
(WebKit::WebContextMenuClient::searchWithGoogle):
* Tools/TestWebKitAPI/Tests/WebCore/DisplayListRecorderTests.cpp:

Canonical link: https://commits.webkit.org/264468@main
